### PR TITLE
Api endpoint

### PR DESCRIPTION
**Pull request by ridaarif98**
- Create an API endpoint to list all posts.
- Create an API endpoint to list all comments for a post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
- Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond to JSON as well.
## Please Note:
- For the current user to access the API endpoints a token has to be exchanged between the server and the client.
- Use the command below to simulate client-side login on your terminal or command line
```
  curl -XPOST -i -H "Content-Type: application/json" -d '{ "user": { "email": "test@gmail.com", "password": "1234test" } }' http://localhost:3000/users/sign_in
```
`replace the email and password values with the one that is registered in the database of the application`
- A token will be generated

token

#### Below are sign_in page without token
image
 
#### Sign_in with token
image

- Display error for Posts without token

image

- Display posts with token

image

 - Display error for Post Comments  without token
 
image

- Display Post comments in JSON with token

showcomment

- Add Comments using the JSON method with token

postcomments

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -22,4 +22,10 @@
   # resources :posts, only: [:destroy]
 
   resources :comments, only: [:destroy]
+
+  namespace :api, defaults: { format: :json } do
+    resources :posts, only: [:index] do
+      resources :comments, only: [:create, :index]
+    end
+  end
```
Comment (oloomoses) [inline config/routes.rb:30]: - [ ] Kindly implement authentication to your API endpoints, both for registration and login.
- Login endpoint - **POST** : `/login`
- Registration endpoint - **POST** : `/register` or `/signup`

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (ridaarif98) [reply config/routes.rb:30]: @oloomoses  thank you for the review. In project requirements, we need to implement it on the post and comment controllers. So first we generate jwt token when login then access this page.

**Comment by faizi2500** [comment]
Hi Reviewer, just to build on the provided review. User authentication is already there. To verify run the following commands. Additionally screenshots have been provided below also. 

- Use the command below to simulate client-side login on your terminal or command line
```
  curl -XPOST -i -H "Content-Type: application/json" -d '{ "user": { "email": "test@gmail.com", "password": "1234test" } }' http://localhost:3000/users/sign_in
```



#### Below are sign_in page without token
image
 
#### Sign_in with token
image

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Faizii, 
## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏

## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**